### ISSUE TITLE
All kinds of player settings

### ISSUE THREAD

**Issue by TrooperCH** [McPlayHDnet/McPlayHD.net#105]
- [ ] (global) toggle if friends can msg you
- [ ] (lobby) visibility option to remove all players in lobby
- [ ] (global) toggle smuggling sam notifications, mainly treasure map found
- [ ] (fastbuilder) toggle for block counter or distance counter
- [ ] (fastbuilder) inventory sorting options
- [ ] (fastbuilder) toggle a bridging mode to hide shop item, overwatch items etc

**Comment by McPlayHD** [comment]
> (global) toggle if friends can msg you
How would you suggest to implement this? We don't have any "friend-settings" menu or something similar.
> (global) toggle smuggling sam notifications, mainly treasure map found
I won't let players disable the notification in chat but maybe the sound because I can see that this gets annoying in the long run.
> (fastbuilder) inventory sorting options
Please specify.
> (fastbuilder) toggle a bridging mode to hide shop item, overwatch items etc
Why? As I remember you can't accidentally join Overwatch during a run anyways (If not then yes... this needs to be fixed) and if you manage to scroll to the shop inventory your attempt will be a failed one anyways.

**Comment by TrooperCH** [comment]
> (global) toggle if friends can msg you
> How would you suggest to implement this? We don't have any "friend-settings" menu or something similar.
add friend settings or something similar, I have heard a certain someone did a whole settings plugin and offered to adapt it

> (fastbuilder) inventory sorting options
> Please specify.
Option to sort the inventory. Hotbar to be more specific. I believe FB originally was BW training and many active players use hotkeys that they would like to also use in practise, but that are not necessarily slot 1 & 2 (especially for telly bridge)

> (fastbuilder) toggle a bridging mode to hide shop item, overwatch items etc
> Why? As I remember you can't accidentally join Overwatch during a run anyways (If not then yes... this needs to be fixed) and if you manage to scroll to the shop inventory your attempt will be a failed one anyways.
It might be failed anyways, but people are complaining because it still is annoying

**Comment by McPlayHD** [comment]
> add friend settings or something similar, I have heard a certain someone did a whole settings plugin and offered to adapt it

This plugin is not at all compatible with friend-specific settings but only settings that change the behaviour of every friend. Or would you be okay if no one could /msg you anymore?

**Comment by McPlayHD** [comment]
We can also add the requested setting of  https://github.com/McPlayHDnet/McPlayHD.net/issues/15 to the list above.

**Comment by TrooperCH** [comment]
> This plugin is not at all compatible with friend-specific settings but only settings that change the behaviour of every friend. Or would you be okay if no one could /msg you anymore?

Yes, this is about the exact reason why people suggest this setting. Servers have this setting e.g. so a person that wants to record a video can disable private messages until he is done with the video.
